test/integration/resourcemanager/systemcomponentsconfig/systemcomponentsconfig_test.go

```diff
@@ -189,17 +201,9 @@ var _ = Describe("SystemComponentsConfig tests", func() {
 				}
 
 				nodes = append(nodes, nonRelevantNode)
-
-				DeferCleanup(func() {
-					Expect(testClient.DeleteAllOf(ctx, &corev1.Node{}, &client.DeleteAllOfOptions{
-						ListOptions: client.ListOptions{
-							LabelSelector: labels.SelectorFromSet(nonRelevantNode.GetLabels()),
-						},
-					})).To(Succeed())
-				})
 			})
 
-			It("should add the node selector and configured tolerations + tolerate node taints", func() {
+			It("should add the node selector and configured tolerations and tolerate taints of existing nodes", func() {
```

Comment:
In this case, there is no existing node that matches the selector, so the test doesn't expect any node taints to be added to the pod.
Hence, I find this description to be confusing.
```suggestion
			It("should add the node selector and configured tolerations", func() {
```